### DISCUSSION TITLE
Is there anyway to use ssh.github.com:443 instead of port 22?

### DISCUSSION THREAD

**Discussion by andloh** [fluxcd/flux2#2733] [Q&A] [answered]
Not all orgs aren't allowed to have port 22 open outbound. Is it possible to configure this today? The inital bootstrap is not the problem, because you can run the process from a machine with internet access. 

Tried the `--hostname` param, but that changes the bootstrap address aswell and the process ofc fails with `ssh.github.com:443` set

**Comment by kingdonb** [comment, marked as answer]
There was an issue in an older version of Flux with regard to setting different ports. If you are on the latest version of Flux today, this should work.

The related issue/PR that closed it: https://github.com/fluxcd/flux2/pull/2231

I understand from your post that initial bootstrapping is not the problem though. So, where does the problem come in? The `--ssh-hostname` tag is for SSH hostname, we were not able to find `--hostname` in `flux create source git`.

If you are using `flux create source github` there is a `--hostname` parameter, but no port parameter. (I think it's a missing flag – going to check this out myself after today's Bug Scrub. Thanks for the report!)

Once Flux is bootstrapped, it should communicate on whatever address is set in the `gotk-sync.yaml` in the GitRepository resource. If you need to set it manually, you should be able to set the address there as `ssh://git@ssh.github.com:443/youruser/yourrepo` and I think that will work.
